src/main/java/org/kiwiproject/registry/util/Paths.java

```diff
@@ -0,0 +1,35 @@
+package org.kiwiproject.registry.util;
+
+import static org.kiwiproject.base.KiwiStrings.f;
+import static org.kiwiproject.registry.util.Ports.findFirstPortPreferSecure;
+
+import lombok.experimental.UtilityClass;
+import org.kiwiproject.registry.model.Port;
+
+import java.util.List;
+
+/**
+ * Utility methods for generating URLs from the service information
+ */
+@UtilityClass
+public class Paths {
```

Comment:
I feel like we should not name this the same as `java.nio.file.Paths`, but cannot think of a better name...

Reply:
Renamed it `ServiceInstancePaths`